src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,163 @@
+            if (executeUpdate < 1) {
+                throw new RuntimeException("Cant add new book with following parameters: " + book);
+            }
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant add new book with following parameters: " + book, e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                book = new Book();
+                Long rowId = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                book.setId(rowId);
+                book.setTitle(title);
+                book.setPrice(price);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("findById failed", e);
+        }
+
+        Optional<Book> result = Optional.ofNullable(book);
+        return result;
+    }
+
+    @Override
+    public Book update(Book book) {
+        if (book == null || book.getId() == null) {
+            throw new RuntimeException("unacceptable data format");
+        }
+        Book currentBook = findById(book.getId()).orElseThrow(() -> new RuntimeException(
+                "There is no object with such id: " + book.getId()));
+
+        if (book.equals(currentBook)) {
+            return book;
+        }
+
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            preparedStatement.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new RuntimeException("update failed", e);
+        }
+
+        currentBook = findById(book.getId()).get();
+        if (currentBook.equals(book)) {
+            return currentBook;
+        }
+        throw new RuntimeException("Update failed");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        if (findById(id).isEmpty()) {
+            throw new RuntimeException(
```

Comment:
The error message 'There is no object with such id' is not very informative. Consider rephrasing to 'No book found with ID: ' + id'.